src/layout/monitor.rs

```diff
@@ -35,6 +35,8 @@ pub struct Monitor<W: LayoutElement> {
     pub workspaces: Vec<Workspace<W>>,
     /// Index of the currently active workspace.
     pub active_workspace_idx: usize,
+    /// ID of the previously active workspace.
+    pub previous_workspace_id: u64,
```

Comment:
It's probably worth it to wrap this in a newtype like `WorkspaceId` to avoid accidentally using it numerically.